### PR TITLE
Subregions

### PR DESCRIPTION
**Pull request by rotsee**
@papandreou Here's an attempt at https://github.com/papandreou/node-cldr/issues/81

You may want to have a look at `fileNamesByTypeAndNormalizedLocaleId`, that is no longer a getter, but will fetch (and cache) only the data type that's going to be used.

### CODE REVIEW

File: lib/cldr.js [hunk 1]
```diff
@@ -1760,6 +1788,46 @@ Cldr.prototype = {
...
+      groupNode => {
+        const type = groupNode.getAttribute('type');
+        const contains = groupNode.getAttribute('contains').split(' ');
+
+        subdivisionContainmentGroups[type] = {
+          type,
+          contains
+        };
+
+        contains.forEach(id => {
```
Comment (papandreou) [inline lib/cldr.js]: I think we should transition to `for ... of` as we go along.

File: lib/cldr.js [hunk 2]
```diff
@@ -1760,6 +1788,46 @@ Cldr.prototype = {
...
+          parentRegionIdByType[id] = type;
+        });
+      }
+    );
+
+    Object.keys(subdivisionContainmentGroups).forEach(type => {
```
Comment (papandreou) [inline lib/cldr.js]: Ditto

File: lib/cldr.js [hunk 3]
```diff
@@ -490,6 +490,34 @@ Cldr.prototype = {
     return territoryDisplayNames;
   },
 
+  /**
+   * Subdivisions are usually subnational administrative entities.
+   * Codes follow the BCP47 standard, e.g. `usca` for California, USA.
+   * Note that these codes are similar but not identical to ISO 3166-2 codes.
+   * Unlike ISO 3166-2, CLDR never reuses a code.
```
Comment (papandreou) [inline lib/cldr.js]: Great to surface this information, could we mention it in the README as well?

File: lib/cldr.js [hunk 4]
```diff
@@ -490,6 +490,34 @@ Cldr.prototype = {
...
+   * cldr.extractSubdivisionDisplayNames('en').dk85
+   */
+  extractSubdivisionDisplayNames(localeId) {
+    this.checkValidLocaleId(localeId);
+    const finder = this.createFinder(
+      this.getPrioritizedDocumentsForLocale(localeId, 'subdivisions')
+    );
+    const subdivisionDisplayNames = {};
+    finder('/ldml/localeDisplayNames/subdivisions/subdivision').forEach(node => {
+      if (node.getAttribute('alt') !== 'variant') {
```
Comment (papandreou) [inline lib/cldr.js]: Other extractors rely on the fact that the “most correct” one always comes first. I think `alt` has other possible values that also indicates less certainty?

File: lib/cldr.js [hunk 5]
```diff
@@ -490,6 +490,34 @@ Cldr.prototype = {
...
+        const subdivisionId = node.getAttribute('type');
+        subdivisionDisplayNames[subdivisionId] = node.textContent;
```
Comment (papandreou) [inline lib/cldr.js]:
```suggestion
        subdivisionDisplayNames[subdivisionId] = subdivisionDisplayNames[subdivisionId] || node.textContent;
```

**Comment by rotsee** [comment]
I fixed your comments. Prettier complains about code style, but I'm not sure how to find out what the code style errors actually are...

**Comment by papandreou** [comment]
It doesn't really make sense to fix the prettier complaints by hand. The standard approach is to install a prettier plugin for your editor and set it up to automatically format on save. Then you never have to worry about it. The vscode and atom plugins have an option to only enable that when the project contains a `.prettierrc` so that you won't do it everywhere :)

If you're not up for that, `node_modules/.bin/prettier --write '**/*.js'` should fix it up.

**Comment by rotsee** [comment]
@papandreou Thanks for the review, I'll leave it to you to merge and release when you see fit!

**Comment by papandreou** [comment]
Great work, released in 5.2.0 :shipit:

**Pull request merged**